modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx

```diff
@@ -0,0 +1,183 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import {Option, Picker} from '@clayui/core';
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayModal from '@clayui/modal';
+import {FieldBase} from 'frontend-js-components-web';
+import {fetch, navigate, objectToFormData} from 'frontend-js-web';
+import React, {useState} from 'react';
+
+type FrontendTokenDefinitionProvider = {
+	name: string;
+	themeId: string;
+};
+
+interface AddStyleBookModalProps {
+	addStyleBookEntryURL: string;
+	closeModal: () => void;
+	frontendTokenDefinitionProviders?: Array<FrontendTokenDefinitionProvider>;
+	namespace: string;
+}
+
+const AddStyleBookModalContent = ({
+	addStyleBookEntryURL,
+	closeModal,
+	frontendTokenDefinitionProviders = [],
+	namespace,
+}: AddStyleBookModalProps) => {
+
+	const [errorMessage, setErrorMessage] = useState<string>('');
+	const [loading, setLoading] = useState(false);
+	const [name, setName] = useState<string>('');
+	const [themeId, setThemeId] =
+		useState<React.Key>(frontendTokenDefinitionProviders[0].themeId);
+	const [showSaveButton, setShowSaveButton] = useState(false);
+
+	const handleFormError = (responseContent: any) => {
+		setErrorMessage(responseContent.error || '');
+	};
+
+	const validateName = (name: string) => {
+		let error = '';
+
+		if (!name.trim().length) {
+			error = Liferay.Language.get('this-field-is-required');
+		}
+
+		setErrorMessage(error);
+
+		setShowSaveButton(error ? true : false);
```

Comment:
I think this would not be necessary to have another state to deal with disabling the button, you can use the error state itself to do this with `Boolean(error)`. This is good because we avoid a double re-render here, in the new versions in React this would not be a problem anymore because it is possible in batch but I believe it is unnecessary for this use case.